### PR TITLE
Increase pages size for github_organization data source

### PR DESCRIPTION
**Pull request by j3parker**
We are looking to use the github_organization data source in part to verify that users from config files (maintained by humans and used by our Terraform code) belong to an organization (something we don't manage via TF) before trying to assign them to teams or as collaborators, etc.

I noticed that the github_organization data source takes ~2 minutes to read for one of our orgs. I think it is related to having to page through the list of repositories and users.

This PR changes the page size for both of those calls from 10 to 100 (the maximum). I did a bit of searching in the repo and I didn't see this coming up before. If there is a reason GitHub would prefer small pages that's fine. I'll link to the docs inline with the diff.

cc @kfcampbell

### CODE REVIEW

File: github/data_source_github_organization.go [hunk 1]
```diff
@@ -96,7 +96,7 @@ func dataSourceGithubOrganizationRead(d *schema.ResourceData, meta interface{})
 	}
 
 	membershipOpts := &github.ListMembersOptions{
-		ListOptions: github.ListOptions{PerPage: 10, Page: 1},
+		ListOptions: github.ListOptions{PerPage: 100, Page: 1},
```
Comment (j3parker) [inline github/data_source_github_organization.go:99]: Default 30, max 100: https://docs.github.com/en/rest/reference/orgs#list-organization-members

**Comment by kfcampbell** [comment]
This change seems reasonable to me! I'm sorry I didn't see/respond to this until now.

**Comment by j3parker** [comment]
Thanks! No worries 😄

**Comment by ben-z** [comment]
> verify that users from config files (maintained by humans and used by our Terraform code) belong to an organization (something we don't manage via TF) before trying to assign them to teams or as collaborators, etc.

@j3parker I'm looking to do the same, but my team didn't have the bandwidth to look into this. Is your implementation public?

**Comment by j3parker** [comment]
> > verify that users from config files (maintained by humans and used by our Terraform code) belong to an organization (something we don't manage via TF) before trying to assign them to teams or as collaborators, etc.
> 
> @j3parker I'm looking to do the same, but my team didn't have the bandwidth to look into this. Is your implementation public?

It isn't, unfortunately. I extracted just those parts and put a demo repo up at [j3parker/github-tf-teams-demo](https://github.com/j3parker/github-tf-teams-demo). The README.md explains the repo layout (note this comes from a larger repo with more stuff (e.g. more usage of this provider, but also AWS infrastructure, code, etc.) the structure may not fit how you do Terraform etc.)

In the README [I linked directly to the bit that filters the team config files to only org members](https://github.com/j3parker/github-tf-teams-demo/blob/c8291b34cbcc6346a27c7deb2e035bc3017bd1df/terraform/modules/github-org-teams/main.tf#L18-L27) in case that's the part you were mostly after.

Note that one part that looks weird is the explicit provider name stuff -- unfortunately I don't remember what that's about and forgot to document it... I found this exchange in our repo though:

image

🤷 

If you have any other questions you can [email me](mailto:jacob@solidangle.ca).

**Comment by ben-z** [comment]
> > > verify that users from config files (maintained by humans and used by our Terraform code) belong to an organization (something we don't manage via TF) before trying to assign them to teams or as collaborators, etc.
> 
> > 
> 
> > @j3parker I'm looking to do the same, but my team didn't have the bandwidth to look into this. Is your implementation public?
> 
> 
> 
> It isn't, unfortunately. I extracted just those parts and put a demo repo up at [j3parker/github-tf-teams-demo](https://github.com/j3parker/github-tf-teams-demo). The README.md explains the repo layout (note this comes from a larger repo with more stuff (e.g. more usage of this provider, but also AWS infrastructure, code, etc.) the structure may not fit how you do Terraform etc.)
> 
> 
> 
> In the README [I linked directly to the bit that filters the team config files to only org members](https://github.com/j3parker/github-tf-teams-demo/blob/c8291b34cbcc6346a27c7deb2e035bc3017bd1df/terraform/modules/github-org-teams/main.tf#L18-L27) in case that's the part you were mostly after.
> 
> 
> 
> Note that one part that looks weird is the explicit provider name stuff -- unfortunately I don't remember what that's about and forgot to document it... I found this exchange in our repo though:
> 
> 
> 
> image
> 
> 
> 
> 🤷 
> 
> 
> 
> If you have any other questions you can [email me](mailto:jacob@solidangle.ca).

Thanks so much! The code and instructions in your repo are very clear. Using ideas from your repo, I was able to solve my use case in a few hours. For future readers--what I did is pretty much the same, except I used `github_team_members` instead of `github_team_membership` to enforce that nobody adds members manually without going through Terraform.

**Pull request merged**